middleware/authentication.go

```diff
@@ -145,17 +135,47 @@ func MakeHTTPOIDCTokenValidationMW(keycloakClient KeycloakClient, audienceRequir
 			ctx = context.WithValue(ctx, cs.CtContextGroups, extractGroups(jot.getGroups()))
 			ctx = context.WithValue(ctx, cs.CtContextIssuerDomain, issuerDomain)
 
-			if err = keycloakClient.VerifyToken(ctx, realm, accessToken); err != nil {
-				logger.Info(ctx, "msg", "Authorization error", "err", err)
-				httpErrorHandler(ctx, http.StatusUnauthorized, errors.New(errorhandler.MsgErrInvalidParam+"."+errorhandler.Token), w)
-				return
-			}
-
 			next.ServeHTTP(w, req.WithContext(ctx))
 		})
 	}
 }
 
+// ValidateOIDCToken ensures the OIDC token given in parameter is valid
+func ValidateOIDCToken(ctx context.Context, accessToken string, keycloakClient KeycloakClient, audienceRequired string, logger log.Logger) (tokenAudience, int) {
+
+	payload, _, err := jwt.Parse(accessToken)
+	if err != nil {
+		logger.Info(ctx, "msg", "Authorization error", "err", err)
+		return nil, http.StatusForbidden
```

Comment:
I would suggest to create 2 kind of errors (Forbidden & Unauthorized). You could then return an error and determine the corresponding HTTP status code in the upper layer. 
The benefit is not really visible here but when you will reuse the method in your component layer it would avoid to have to manipulate HTTP code at this layer.

Reply:
The HTTP response is not written in another layer... it is written directly in authentication.go when calling httpErrorHandler(...)

Reply:
This method will be called by signature-service at component layer. That's one of the reason why we did this refactoring.